docs/boring-bomb-architecture-doc.md

```diff
@@ -0,0 +1,115 @@
+6. Finally, a user with the Trust Assembly browser extension visits a website, and the extension uses the API to fetch a headline for the article they’re reading. If no `Article` exists in our DB, then nothing will be changed about their website’s article. If one does exist, then the extension will use the unbiased headline from the `Article`
+
+## 🧑‍💼 Business Objects and Their Relations
+
+1. `Datasource`
+    1. id
+    2. parent_id (if the url to the same article has changed over time, then the new `Datasource` will have the prior `Datasource` as its parent_id)
+    3. base_url (i.e. [`https://extelligence.substack.com/archive`](https://extelligence.substack.com/archive))
+    4. owner_id
+    5. time_last_scraped
+2. `Article`
+    1. id
+    2. original_text
+    3. headline_text
+    4. edit_pairs (this will be an array of [original_text, edited text] tuples, where `original_text` is the original sentence/paragraph/section of text, and `edited_text` is the text to replace the `original_text` with)
+    5. parent_id (if the article original_text has changed over time, then there will be a new `Article` whose parent_id points to the previous `Article`)
+    6. datasource_id
+    7. url
+    8. time_created
+    9. time_last_updated
+3. `DatasourceOwner` (this is not for auth, but rather for the prompt we’ll use to influence the voice of the generated Headline)
+    1. id
+    2. prompt_text
+    3. prompt_text_arguments (the prompt text is expected )
+    4. name
+    5. email_contact
+    6. website
+
+## 🧩 Deno API Server Definitions (For Browser Extension)
+
+This is very simple, because at the end of the day all boring bomb is trying to do is generate unbiased headlines for articles, and make those headlines available to the user.
+
+### Client Boring Bomb API
+
+1. `GET /headline`
+
+## ❓Remaining Questions
+
+1. How will scraping work be distributed amongst the workers?
+2. How will worker and API logs be collected, queried, and viewed?
+3. If the URL for a site changes, how do we determine the new `Datasource` which will have the `Datasource` with the previous URL as its `parent_id`?
+4. How much does an `Article` need to mutate before we create a new `Article` that references the old `Article` as its `parent_id`?
+5. How will Boring Bomb acquire and maintain access to paywalled `Datasources` like NYT and other paid subscription articles?
+6. Should we include images in our system? Storing image data will greatly increase the amount of space we need to pay for, and require a different type of storage (i.e. blob storage like S3 or Cloudflare’s R2), but images may contain important information. One way to get the best of both worlds may be to simply store image caption data if it exists, or have AI do a synopsis of the image and just storage that. The question to answer is: what benefit does storing image data give to Trust Assembly?
```

Comment:
Is this about images from the articles? Those should just be URLs in the `img` tag. Would editors want to add their own images?

Reply:
Ah i wasn't thinking of this from the editors point of view, mostly because I'm not thinking much about editors in this early MVP stage.

I was thinking of it from Boring Bomb's perspective, where having access to an article's images might be critical for analyzing the meaning of the article and ultimately creating a neutral headline. For example if an article has tons of spin in it, and references a graph from a reputable source (science.org) that contains true information, then the headline should reflect that.

This feels like a concern for beyond the MPV, because it's an optimization (specifically optimizing the quality of the headlines). I say we punt this for post MVP.

I'll make a section for features/components we considered, but will not include in this MVP architecture doc.